storage/src/main/java/io/atomix/storage/journal/MappedJournalSegmentReader.java

```diff
@@ -84,12 +84,16 @@ public long getNextIndex() {
   @Override
   public void reset(long index) {
     reset();
-    Position position = this.index.lookup(index - 1);
-    if (position != null) {
+
+    final Position position = this.index.lookup(index - 1);
```

Comment:
Oh yes more duplicated code I love to see this...not

Reply:
:sweat_smile: Before we start refactoring more of this, I just hope we can drop this journal

Reply:
would be nice